File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        int qui = n;
        String str = "";
        int remain = 0;
        while(qui>1){
            qui = n/k;
            remain = n%k;
            if(remain==0) str+=" ";
            else str+= remain;
            
            n = qui;
            if(qui==1) str+=1;
        }
        StringBuffer sb = new StringBuffer(str);
        String num = sb.reverse().toString();
        StringTokenizer st = new StringTokenizer(num, " ");
        while(st.hasMoreTokens()){
            Long prime= Long.parseLong(st.nextToken());
            if(prime<=1)continue;
            int half =(int)Math.sqrt(prime);
            
            boolean isPrime  =true;
            for(int i=2; i<=half; i++){
                if(prime%i==0) {
                    isPrime = false;   
                    
                    break;
                }
            }
            if(isPrime) answer++;
            
        }
        return answer;
    }
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;
bool isPrime(long long num,int n){
    if(num<2) return false;
    for(long long i =2; i*i<=num;i++){
        if (num % i == 0) return false;    
    }
    return true;
}
int solution(int n, int k) {
    int answer = 0;
    string str;
    long long temp_n = n;
    while(n){
        str+= n%k+'0';
        n = n/k;
    }
    reverse(str.begin(), str.end());

    string temp="";
    for(char c : str){
        if(c=='0'){
            if(!temp.empty()){
                if(isPrime(stoll(temp),temp_n)) answer++;
            }
            temp="";
        }else{
            temp+=c;
        }
    }
    if(!temp.empty())if(isPrime(stoll(temp),temp_n)) answer++;
    
    return answer;
}